pandas/_libs/tslibs/period.pyx

```diff
@@ -1416,6 +1416,32 @@ cdef class _Period(object):
 
     @property
     def weekday(self):
+        """
+        Get day of the week that a Period falls on.
```

Comment:
As this is a property and not a function, no need to start with a verb, you can start with "Day of the week...".

Also, as `Period` here is the pandas object, you can use backticks to highlight it.

I think in the extended summary you could explain that `0` means `Monday`...